File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int counter = 0;
        int current = 0;
        for (int i = 0; i < nums.length; ++i) {
          if (counter == 0) {
            current = nums[i];
            counter++;
          } else if (nums[i] == current) {
            counter++;
          } else {
            counter--;
          }
        }

        return current;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate = 0
        counter = 0
        for i in range(0, len(nums)):
            if counter == 0:
                candidate = nums[i]
                counter += 1
            elif nums[i] == candidate:
                counter += 1
            else:
                counter -= 1
        return candidate
        